tests/pods_test.go

```diff
@@ -37,6 +38,13 @@ type Event struct {
 func TestPod(testT *testing.T) {
 	// https://stack-rox.atlassian.net/browse/ROX-6631
 	// - the process events expected in this test are not reliably detected.
+
+	if os.Getenv("COLLECTION_METHOD") == "NO_COLLECTION" {
+		testT.Log("Skipping TestPod because env var \"COLLECTION_METHOD\" is set to \"NO_COLLECTION\"\n" +
+			"This is expected to only happen when Sensor version is 3.74.x (support exception within compatibility tests)")
```

Comment:
We need to decide what we do here. In this block, you print the log and then return. This is literally the same as using `testT.Skip`, but marks this test as passed, whereas `Skip` marks it as skipped. This is important when analyzing test results using JUnit, so we must use the correct method.

But there is also another issue here - if we skip this test, then the `if` clause in line 72 makes no sense as it would always be evaluated to false.

(blocker for merging) 🛑  You must decide: skip parts or skip the entire test, but not both together.